src/bank_api.cpp

```diff
@@ -0,0 +1,171 @@
+#include "../headers/bank_api.h"
+
+using namespace Pistache;
+
+
+BankAPI::BankAPI(Address addr) : httpEndpoint(std::make_shared<Http::Endpoint>(addr)) {}
+
+void BankAPI::init(size_t thr) {
+    auto opts = Http::Endpoint::options().threads(static_cast<int>(thr));
+    httpEndpoint->init(opts);
+    setupRoutes();
+}
+
+void BankAPI::start() {
+    cout << "Server is running on port 8080..." << endl;
+    httpEndpoint->setHandler(router.handler());
+    httpEndpoint->serve();
+}
+
+void BankAPI::shutdown() {
+    httpEndpoint->shutdown();
+}
+
+void BankAPI::setupRoutes() {
+    using namespace Rest;
+
+    // Route to create an account
+    Routes::Post(router, "/bank/account", Routes::bind(&BankAPI::createAccount, this));
+
+    // Route to consult an account
+    Routes::Get(router, "/bank/account/:accountID", Routes::bind(&BankAPI::consultAccount, this));
+
+    // Route to consult the balance of an account
+    Routes::Get(router, "/bank/account/:accountID/balance", Routes::bind(&BankAPI::consultAccountBalance, this));
+
+    // Route to credit an account
+    Routes::Put(router, "/bank/account/:accountID/credit", Routes::bind(&BankAPI::creditAccount, this));
+
+    // Route to debit an account
+    Routes::Put(router, "/bank/account/:accountID/debit", Routes::bind(&BankAPI::debitAccount, this));
+
+    // Route to transfer between accounts
+    Routes::Put(router, "/bank/account/transfer", Routes::bind(&BankAPI::transferBetweenAccounts, this));
+
+    // Route to apply income to an account
+    Routes::Put(router, "/bank/account/income", Routes::bind(&BankAPI::income, this));
+
+}
+
+// TODO: enhance logic to remove code repetitions
+void BankAPI::createAccount(const Rest::Request& request, Http::ResponseWriter response) {
+    json payload = json::parse(request.body());
+    
+    int accountType = payload["type"];
+    int accountID = payload["id"];
+    double accountOpeningBalance = payload["openingBalance"].get<double>();
```

Comment:
Esse `openingBalance` é optativo passar no body? Porque caso seja uma conta do tipo bônus, o usuário não vai informar isso.

Reply:
ele só é utilizado quando a conta é normal ou poupança. nesses casos, se ele não for informado, pensei em duas soluções:
1. setar pra 0
2. retornar bad request

o que seria melhor?